packages/flutter_tools/test/integration.shard/break_on_framework_exceptions_test.dart

```diff
@@ -0,0 +1,737 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'package:file/file.dart';
+
+import '../src/common.dart';
+import 'test_data/project.dart';
+import 'test_driver.dart';
+import 'test_utils.dart';
+
+void main() {
+  Directory tempDir;
+
+  setUp(() async {
+    tempDir = createResolvedTempDirectorySync('break_on_framework_exceptions.');
+  });
+
+  tearDown(() async {
+    tryToDelete(tempDir);
+  });
+
+  testWithoutContext('breaks when AnimationController listener throws', () async {
+    final TestProject project = TestProject(
+      r'''
+      AnimationController(vsync: TestVSync(), duration: Duration.zero)
+        ..addListener(() {
+          throw 'AnimationController listener';
+        })
+        ..forward();
+      '''
+    );
+    await project.setUpIn(tempDir);
+    final FlutterTestTestDriver flutter = FlutterTestTestDriver(tempDir);
+    await flutter.test(withDebugger: true, pauseOnExceptions: true);
+    await flutter.waitForPause();
+
+    expect((await flutter.getSourceLocation()).line, equals(project.lineContaining(project.test, "throw 'AnimationController listener';")));
```

Comment:
nit: equals doesn't really do much here besides make the lines longer. Also, I would split these lines up a bit to make this easier to read. perhaps something like:

```dart
final IDonTKnowWhatType sourceLocation = await flutter.getSourceLocation();
expect(sourceLocation.line, project.lineContaining(project.test, "throw 'AnimationController listener';"));

````